lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -399,6 +401,19 @@ instance Buildable TxStatus where
         InLedger -> "in ledger"
         Invalidated -> "invalidated"
 
+instance FromText TxStatus where
+    fromText txt = case txt of
+        "Pending" -> Right Pending
+        "InLedger" -> Right InLedger
+        "Invalidated" -> Right Invalidated
+        _ ->
+            Left . TextDecodingError $ "not a valid value: " <> show txt
```

Comment:
A better error message would be to point out to what are the expected values (as we do in other places).

Reply:
done